### PR TITLE
[cmd] Fixed #1329 create a command for the IP monitors

### LINKED ISSUE #1329: Tool to monitor peers.txt IP addresses
Create a tool (either a new cli command or a new `cmd/` binary) that takes in a list of peers (ip:ports, newline separated, skipping comments and empty lines (a function exists somewhere in the `daemon` or `pex` package to parse this format)).  The tool connects to each of the peers, waits for the introduction packet (or times out) and produces a report with the status of the peer (unreachable, reachable, sent_introduction, introduction_parameters).  Introduction_parameters were added in v0.25.0 so will be absent for earlier peer versions.

The initial purpose is to check the status of peers in https://downloads.skycoin.net/blockchain/peers.txt

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #1329

Changes:
-Create a new `monitor` command with two flags,
 `-f` to give the address the file `peers.txt` that should be checked, in case of not calling it the default value is `skycoin/skycoin/peers.txt`
 `-t` to define the maximum waiting time in seconds for checking the IP, the default value is 5.

Does this change need to mentioned in CHANGELOG.md?
Yes

### CODE REVIEW

File: cmd/monitor/monitor.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		peers[pex] = bufferpex
+
+	}
+
+	return output, peers
+
+}
+func main() {
+
+	urlPeer := flag.String("f", urlPeers, "Url the file peer.txt")
```
Comment (olemis) [inline cmd/monitor/monitor.go]: You should have a configuration struct with fields initially set to default values and populated by `flag` afterwards. Parsing should happen inside a method (receiver) named `ParseConfig` .

Comment (gz-c) [reply cmd/monitor/monitor.go]: Thats not necessary for a simple cli

Comment (olemis) [reply cmd/monitor/monitor.go]: 👍

Comment (olemis) [reply cmd/monitor/monitor.go]: @gz-c I'll approve changes then . Please review .

Comment (olemis) [reply cmd/monitor/monitor.go]: Oh ! Noticed something ...

File: cmd/monitor/monitor.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	"strings"
+	"time"
+
+	"github.com/skycoin/skycoin/src/util/logging"
+)
+
+var (
+	logger = logging.MustGetLogger("monitor")
+	// ErrPeerlistFull is returned when the Pex is at a maximum
+	ErrPeerlistFull = errors.New("Peer list full")
```
Comment (olemis) [inline cmd/monitor/monitor.go]: Is this used somewhere in command source code ?

File: cmd/monitor/monitor.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	// ErrInvalidAddress is returned when an address appears malformed
+	ErrInvalidAddress = errors.New("Invalid address")
```
Comment (olemis) [inline cmd/monitor/monitor.go]: Isn't it possible to reuse the error objects defined in `src/` folder ? e.g.

```sh
$ grep -nr 'ErrInvalidAddress = ' src
src/daemon/pex/pex.go:44:	ErrInvalidAddress = errors.New("Invalid address")
```

File: cmd/monitor/monitor.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+	ErrBlacklistedAddress = errors.New("Blacklisted address")
+
+	whitespaceFilter = regexp.MustCompile(`\s`)
+
+	dialTimeout = 5 * time.Second
+	urlPeers    = os.Getenv("GOPATH") + "/src/github.com/skycoin/skycoin/peers.txt"
+)
+
+// validateAddress returns a sanitized address if valid, otherwise an error
+func validateAddress(ipPort string) (string, error) {
```
Comment (olemis) [inline cmd/monitor/monitor.go]: if this function is already defined elsewhere in `src/` isn't it better to export it for reuse ?

```
$ grep -nr 'func validateAddress' src/
src/daemon/pex/pex.go:64:func validateAddress(ipPort string, allowLocalhost bool) (string, error) {
```

File: cmd/monitor/monitor.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+	output, outputPeers := getPeers()
+
+	outputJSON, err := json.MarshalIndent(outputPeers, "", "     ")
+	if err != nil {
+		fmt.Println("Error formating wallet to JSON. Error:", err)
+		os.Exit(1)
+	}
+
+	fmt.Println(string(outputJSON))
```
Comment (therealssj) [inline cmd/monitor/monitor.go]: use log

File: cmd/monitor/monitor.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+)
+
+var (
+	logger      = logging.MustGetLogger("monitor")
+	dialTimeout = 5 * time.Second
+	urlPeers    = getPathGo() + "/src/github.com/skycoin/skycoin/peers.txt"
+)
+
+func getPathGo() string {
+	gopath := os.Getenv("GOPATH")
```
Comment (olemis) [inline cmd/monitor/monitor.go]: Use `os.LookupEnv` instead of `os.GetEnv`

**Comment by gz-c** [comment]
#2268

**Pull request closed without merging**